src/rgw/rgw_zone_types.h

```diff
@@ -542,6 +542,44 @@ struct RGWZoneGroupPlacementTierS3 {
 };
 WRITE_CLASS_ENCODER(RGWZoneGroupPlacementTierS3)
 
+enum GlacierRestoreTierType {
+  Standard = 0,
+  Expedited = 1,
+};
+
+struct RGWZoneGroupTierS3Glacier {
+#define DEFAULT_GLACIER_RESTORE_DAYS 1
+  uint64_t glacier_restore_days = DEFAULT_GLACIER_RESTORE_DAYS;
+  GlacierRestoreTierType glacier_restore_tier_type{Standard};
+
+  int update_params(const JSONFormattable& config);
+  int clear_params(const JSONFormattable& config);
+
+  void encode(bufferlist& bl) const {
+    ENCODE_START(1, 1, bl);
+    encode(glacier_restore_days, bl);
+    encode(glacier_restore_tier_type, bl);
+    ENCODE_FINISH(bl);
+  }
+
+  void decode(bufferlist::const_iterator& bl) {
+    DECODE_START(1, bl);
+    decode(glacier_restore_days, bl);
+    uint32_t it;
+    decode(it, bl);
+    glacier_restore_tier_type = (GlacierRestoreTierType)it;
```

Comment:
if you restrict the enum to 8 bits with `enum GlacierRestoreTierType : uint8_t`, you can encode/decode it directly